### ISSUE TITLE
lint/busySelect_checker.go: false positive if for loop blocks outside of select

### ISSUE THREAD

**Issue by aaron42net** [go-critic/go-critic#631]
BusySelect complains about an unconditional for loop containing a non-blocking select, but doesn't check to see whether the other statements in the for loop outside of the select are blocking.

For example, it complains about this, which is a false positive:

```
        for {
                msgs <- l
                select {
                case <-done:
                        return
                default:
                }
        }
```

Is it reasonable to have hasBlockingStmt() check all statements in the for loop, not just the select?

**Comment by cristaloleg** [comment]
Oh, that's a good example, thanks. Will be fixed during current milestone 😉
